narwhals/_duration.py

```diff
@@ -0,0 +1,50 @@
+"""Tools for working with the Polars duration string language."""
+
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING
+from typing import Literal
+from typing import cast
+from typing import get_args
+
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
+__all__ = ["IntervalUnit", "parse_interval_string"]
+
+IntervalUnit: TypeAlias = Literal["ns", "us", "ms", "s", "m", "h", "d", "mo", "q", "y"]
+
+PATTERN_INTERVAL: re.Pattern[str] = re.compile(
+    r"^(?P<multiple>\d+)(?P<unit>ns|us|ms|mo|m|s|h|d|q|y)$"
```

Comment:
I saw this maybe an hour or two after I wrote (https://github.com/narwhals-dev/narwhals/pull/2465#discussion_r2070531991)

- https://discuss.python.org/t/proposal-add-z-as-a-synonym-for-z-in-python-res-for-standardization/90378

```suggestion
    r"^(?P<multiple>\d+)(?P<unit>ns|us|ms|mo|m|s|h|d|q|y)\Z"
```

Reply:
Not sure I follow the suggestion, sorry - is `\Z` preferred for some reason? I've never seen it, I've always only seen `^` and `$`. I'd suggest keeping this one as-is if it's ok

Reply:
@MarcoGorelli did you read the thread I linked? 

It is preferred and will be getting a new synonym (`\z`) - since (`$`) is commonly misused.
I linked the thread because **I did the exact misuse** 😅

Reply:
No worries @MarcoGorelli!

You're only 2 days behind me 😄